### PR TITLE
Move big data generator to a separate module

### PR DESCRIPTION
**Pull request by jlowe**
This moves the data generator out of the integration_tests module and into its own module, making it simpler to pull in as a dependency and more focused if we decide to publish the build artifact later.

### CODE REVIEW

**Comment by jlowe** [comment]
build

File: datagen/README.md [hunk 1]
```diff
@@ -27,13 +26,13 @@ If you are building with a jdk version that is not 8, you will need to add in th
 corresponding profile flag `-P<jdk11|jdk17>`
 
 After this the jar should be at 
-`target/rapids-4-spark-integration-tests_2.12-$PLUGIN_VERSION-spark$SPARK_VERSION.jar`
+`target/datagen_2.12-$PLUGIN_VERSION-spark$SPARK_VERSION.jar`
```
Comment (abellina) [inline datagen/README.md:29]: should this jar be something like `rapids-4-spark-datagen`? I guess we are not publishing it, so maybe that's not such a big deal (feels a bit generic)

**Review by abellina — APPROVED** [review]
I just had one question that doesn't block this

File: datagen/README.md (hunk 1 above)
Comment (jlowe) [reply datagen/README.md:29]: I originally had it named that way, but there's nothing in it specific to the RAPIDS Accelerator for Apache Spark.  It only uses Apache Spark interfaces and doesn't require the RAPIDS Accelerator.

**Review by revans2 — APPROVED** [review]

**Comment by revans2** [comment]
build

**Pull request merged**